### ISSUE TITLE
Tomboy on Windows no longer starts, --debug hints problem with XML file reading

### ISSUE THREAD

**Issue by erik-brangs** [tomboy-notes/tomboy#56]
I'm using Tomboy on Windows. It worked fine yesterday but today Tomboy no longer starts. So I upgraded to the latest Tomboy version which didn't fix it. Then I updated the dependencies, which didn't help either.

Then I ran
`"C:\Program Files (x86)\Tomboy\Tomboy.exe" --debug`
 from the terminal. The output is:

Unbehandelte Ausnahme: System.Xml.XmlException: Das Stammelement ist nicht vorhanden.
[roughly translates to unhandled exception - root element missing]
   bei System.Xml.XmlTextReaderImpl.Throw(Exception e)
   bei System.Xml.XmlTextReaderImpl.ParseDocumentContent()
   bei System.Xml.XmlTextReaderImpl.Read()
   bei System.Xml.XmlLoader.Load(XmlDocument doc, XmlReader reader, Boolean preserveWhitespace)
   bei System.Xml.XmlDocument.Load(XmlReader reader)
   bei System.Xml.XmlDocument.Load(String filename)
   bei Tomboy.XmlPreferencesClient..ctor() in C:\MyDocs\dev\tomboy\tomboy-github\Tomboy\XmlPreferencesClient.cs:Zeile 39.
   bei Tomboy.WindowsFactory.CreatePreferencesClient() in C:\MyDocs\dev\tomboy\tomboy-github\Tomboy\WindowsFactory.cs:Zeile 18.
   bei Tomboy.Services.get_Preferences() in C:\MyDocs\dev\tomboy\tomboy-github\Tomboy\Services.cs:Zeile 60.
   bei Tomboy.Preferences.get_Client() in C:\MyDocs\dev\tomboy\tomboy-github\Tomboy\Preferences.cs:Zeile 64.
   bei Tomboy.Preferences.Get(String key) in C:\MyDocs\dev\tomboy\tomboy-github\Tomboy\Preferences.cs:Zeile 176.
   bei Tomboy.TomboyTray.MakeTrayNotesMenu() in C:\MyDocs\dev\tomboy\tomboy-github\Tomboy\Tray.cs:Zeile 396.
   bei Tomboy.TomboyTray..ctor(NoteManager manager) in C:\MyDocs\dev\tomboy\tomboy-github\Tomboy\Tray.cs:Zeile 382.
   bei Tomboy.TomboyTrayIcon..ctor(NoteManager manager) in C:\MyDocs\dev\tomboy\tomboy-github\Tomboy\Tray.cs:Zeile 149.
   bei Tomboy.Tomboy.StartTrayIcon() in C:\MyDocs\dev\tomboy\tomboy-github\Tomboy\Tomboy.cs:Zeile 197.
   bei Tomboy.Tomboy.Main(String[] args) in C:\MyDocs\dev\tomboy\tomboy-github\Tomboy\Tomboy.cs:Zeile 156.

How can I fix this? The error message neglects to mention which XML file is affected so I don't know where to start.

PS: I hope this is the correct place to ask. I didn't see any statement about preferred support channels. Github has a new functionality for that via the SUPPORT file.

**Comment by alex-ter** [comment]
This is a correct place to ask for help, yes. We don't use the SUPPORT file functionality, but the main readme describes ways to get help.

As for the crash - looks like Tomboy fails to load the XML configuration file, it does exist, but the XML Reader class doesn't like it. It should be in `%APPDATA%\Tomboy\config\prefs.xml`, please check it to see if it's corrupted, looks like it is. Any unclean shutdowns/resets between now and when it worked the last time? Per my observations that's number one (and honestly only) reason for such things, be it notes or preferences.

**Comment by erik-brangs** [comment]
> Any unclean shutdowns/resets between now and when it worked the last time?

Yes, IIRC Windows 10 complained about some kind of error during shutdown. I think there was this blueish screen which claimed that information was collected about `REFERENCE_BY_POINTER`. 

> As for the crash - looks like Tomboy fails to load the XML configuration file, it does exist, but the XML Reader class doesn't like it. It should be in %APPDATA%\Tomboy\config\prefs.xml, please check it to see if it's corrupted, looks like it is.

It exists but it's completely empty.

What should I do? Delete it?

**Comment by alex-ter** [comment]
If you don't have any backups of that one - yes  it should be recreated with defaults.

**Comment by erik-brangs** [comment]
Thanks. After deleting the file, everything worked again. I've since restored the settings from backup.

**Comment by alex-ter** [comment]
Ok, great you have it working and thanks for reporting back. Let's keep this one open for a while - I'd like to enhance that part of code to report some more meaningful error, rather than just throwing an exception.
